src/vs/vscode.d.ts

```diff
@@ -4354,7 +4354,7 @@ declare module 'vscode' {
 		 * @param uri A resource identifier.
 		 * @param diagnostics Array of diagnostics or `undefined`
 		 */
-		set(uri: Uri, diagnostics: Diagnostic[] | undefined): void;
+		set(uri: Uri, diagnostics: ReadonlyArray<Diagnostic> | undefined): void;
```

Comment:
Opting to use `ReadonlyArray<Diagnostic>` instead of `readonly Diagnostic[]` in our public api so that extension using older TS version do not get a compile error

Reply:
Tricky - I think `readonly Foo[]` reads a lot nicer than `ReadonlyArray`. We had a similar call when making the API strict null aware (see https://github.com/microsoft/vscode/issues/6907). I think back then we made those chances a good while after TypeScript had enabled strict null

Reply:
Yes, this syntax is still pretty new so I don't think its worth breaking people. Intellisense will also display `ReadonlyArray<T>` as `readonly T[]` in versions of TS that support it